### PR TITLE
fix(post): allow authors to edit and delete their posts

### PR DESCRIPTION
**Pull request by xHyroM**
## What does this PR do?

This PR addresses the issue of post management by authors. With these changes, authors are now able to edit and delete their own posts.

## How did you verify your code works?

by testing it ;)

### CODE REVIEW

**Comment by Molenusaczech** [comment]
Looks good to me

File: endpoint/post.py [hunk 1]
```diff
@@ -98,14 +102,18 @@ def on_delete(self, req, resp, id):
...
                 resp.status = falcon.HTTP_400
                 return
 
             post = session.query(model.Post).get(id)
             if post is None:
                 resp.status = falcon.HTTP_404
                 return
+            
+            if (not user.is_org() and post.author != user.id):
+                resp.status = falcon.HTTP_400
```
Comment (esoadamo) [inline endpoint/post.py:115]: please use [HTTP Code 403 (Forbidden)](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_client_errors), the code 400 was used by legacy backend code and I would like to change it to the newer format :slightly_smiling_face: 

```suggestion
                resp.status = falcon.HTTP_403
```

**Review by esoadamo — CHANGES_REQUESTED** [review]
Hey! Thanks for the contribution, great job! Just a one small change an we can merge it ASAP!

**Review by esoadamo — APPROVED** [review]

**Comment by esoadamo** [comment]
Deployed :slightly_smiling_face:

**Pull request merged**